animatplot/xanim.py

```diff
@@ -0,0 +1,61 @@
+import animatplot as amp
+import xarray as xr
+
+
+def animated_plot(da, anim_over='__guess__', x='x', y='y', plot_type='imshow', fps=10):
+    """
+    Function to plot an animated plot from an xarray DataArray.
+
+    The intention is that this would form the basis of a xr.DataArray.plot.animate() method.
+    """
+
+    if not isinstance(da, xr.DataArray):
+        raise TypeError('First argument must be an xarray DataArray object')
+
+    timeline, evolving_coord = _timeline_from_coords(da.coords, anim_over, fps)
+    t_axis = da.dims.index(evolving_coord)
+
+    if len(da.dims) is not 3:
+        raise NotImplementedError('Currently only plots 2D dataarrays')
+
+    if plot_type is 'imshow':
+        # TODO ideally the blocks method should call the xarray plotting method da.plot.imshow()
+        print(da.values.shape)
+        print(t_axis)
+        block = amp.blocks.Imshow(da.values, t_axis=t_axis)
+    else:
+        # TODO deal with other kinds of plot (lines, quiver...) here
+        raise NotImplementedError('Currently only plots using imshow')
+
+    # TODO if we can't call the xarray plotting method directly then add titles, axes labels etc here
+    anim = amp.Animation([block])
+
+    return anim, block, timeline
```

Comment:
I really like the way you return things here. It bodes well for future functions like this. But, I sort of think it should be a list of blocks (because future convenience functions could return multiple blocks). OTOH, advanced tuple unpacking
```
anim, *blocks, timeline = ...
```
is a thing. OTOH,

```
anim, [block1, block2], timeline = ...
```
is also possible. That's a tough decision. Keep with this and I'll make a decision befor 0.3 release.

Reply:
I personally like the 

    anim, [block1, block2], timeline = ...

variant more, but this is something that would be easy to change anyway.

Reply:
Easy to change, but annoying to change after release. Ya, make it return a list of blocks. It will be easier to explain in the docs too.